### DISCUSSION TITLE
Cannot use ${{}} in contexts in shell command

### DISCUSSION THREAD

**Discussion by jbcpollak** [community/community#27120] [Actions]
<p>I’m trying to use a custom shell command to set an rc file for bash. What I want to do is:</p>
<pre><code>shell: bash --rcfile ${{ github.workspace }}/.github/workflows/bashrc -eo pipefail {0}
</code></pre>
<p>or use a short form like this:</p>
<pre><code>env:
SHELL: bash --rcfile ${{ github.workspace }}/.github/workflows/bashrc -eo pipefail {0}
...
# sometime later
- name: foo
shell: ${{ env.SHELL }}
</code></pre>
<p>In both cases it appears the context cannot be resolved within the shell parameter, as I get this error:</p>
<ul>
<li>Your workflow file was invalid: The pipeline is not valid. .github/workflows/ci.yaml (Line: 23, Col: 14): Unrecognized named-value: ‘env’. Located at position 1 within expression: env.SHELL,.github/workflows/ci.yaml (Line: 40, Col: 14): Unrecognized named-value: ‘env’. Located at position 1 within expression: env.SHELL</li>
</ul>
<p>I get a similar error for the <code>github</code> context when I try the other method as well.</p>
<p>BTW, it would be handy to have the <code>shell:</code> option at the job level so we can set a default instead of just per-step.</p>

**Comment by BrightRan** [comment]
<p>When using the <strong>shell</strong> keyword to specify and set a shell, currently we seems are not able to use variables (such as environment variables, matrix parameters) in the value of the <strong>shell</strong> keyword.</p>
<p>In addition, the default workspace of the current running job generally is <strong>${{ github.workspace }}</strong>. Because the RC file you want to set is contained in the workspace, maybe you can try to only specify the <strong>relative path</strong> of the RC file instead of the full absolute path.</p>
<p>For example:</p>
<pre><code>shell: bash --rcfile .github/workflows/bashrc -eo pipefail {0}
</code></pre>
<p>OR</p>
<pre><code>shell: bash --rcfile ./.github/workflows/bashrc -eo pipefail {0}
</code></pre>

**Comment by alun** [comment, marked as answer]
<p>Custom <code>rcfile</code> setting that worked for me with default bash shell (no override):</p>
<pre><code class="lang-auto">env:
  BASH_ENV: .github/workflows/bashrc
</code></pre>

**Comment by jbcpollak** [comment]
<p>Thanks <a class="mention" href="/alun">@alun</a> ! I worked around this in my build mainly by duplicating the steps we’d take in the rc file, but this looks like the right solution to me.</p>
